packages/next/build/jest/jest.ts

```diff
@@ -157,7 +157,7 @@ export default function nextJest(options: { dir?: string } = {}) {
           // To match Next.js behavior node_modules is not transformed, only `transpiledPackages`
           ...(transpiled
             ? [
-                `/node_modules/(?!(${transpiled}))/`,
+                `/node_modules/(?!.pnpm)(?!(${transpiled})/)`,
```

Comment:
This ensures we discard any matches that are in the `.pnpm` directory (handled below) or that match our transpiled package names.